src/api/index.ts

```diff
@@ -0,0 +1,395 @@
+  GpQuoteErrorDetails,
+} from './errors/QuoteError'
+import { toErc20Address } from '../utils/tokens'
+import { FeeQuoteParams, PriceInformation, PriceQuoteParams, SimpleGetQuoteResponse } from '../utils/price'
+
+import { DEFAULT_NETWORK_FOR_LISTS } from '../constants/lists'
+import { GAS_FEE_ENDPOINTS } from '../constants'
+import { ZERO_ADDRESS } from '../constants'
+import {
+  GasFeeEndpointResponse,
+  GetOrdersParams,
+  GetTradesParams,
+  OrderCancellationParams,
+  OrderID,
+  OrderMetaData,
+  PriceStrategy,
+  ProfileData,
+  TradeMetaData,
+} from './types'
+import { SdkError } from '../utils/common'
+
+function getGnosisProtocolUrl(isDev: boolean): Partial<Record<ChainId, string>> {
+  if (isDev) {
+    return {
+      [ChainId.MAINNET]: 'https://barn.api.cow.fi/mainnet/api',
+      [ChainId.RINKEBY]: 'https://barn.api.cow.fi/rinkeby/api',
+      [ChainId.XDAI]: 'https://barn.api.cow.fi/xdai/api',
+    }
+  }
+
+  return {
+    [ChainId.MAINNET]: 'https://api.cow.fi/mainnet/api',
+    [ChainId.RINKEBY]: 'https://api.cow.fi/rinkeby/api',
+    [ChainId.XDAI]: 'https://api.cow.fi/xdai/api',
+  }
+}
+
+function getProfileUrl(isDev: boolean): Partial<Record<ChainId, string>> {
+  if (isDev) {
+    return {
+      [ChainId.MAINNET]: 'https://barn.api.cow.fi/affiliate/api',
+    }
+  }
+
+  return {
+    [ChainId.MAINNET]: 'https://api.cow.fi/affiliate/api',
+  }
+}
+
+const STRATEGY_URL_BASE = 'https://raw.githubusercontent.com/gnosis/cowswap/configuration/config/strategies'
+
+function getPriceStrategyUrl(): Record<ChainId, string> {
+  return {
+    [ChainId.MAINNET]: STRATEGY_URL_BASE + '/strategy-1.json',
+    [ChainId.RINKEBY]: STRATEGY_URL_BASE + '/strategy-4.json',
+    [ChainId.XDAI]: STRATEGY_URL_BASE + '/strategy-100.json',
+  }
+}
+
+const UNHANDLED_QUOTE_ERROR: GpQuoteErrorObject = {
+  errorType: GpQuoteErrorCodes.UNHANDLED_ERROR,
+  description: GpQuoteErrorDetails.UNHANDLED_ERROR,
+}
+
+const UNHANDLED_ORDER_ERROR: ApiErrorObject = {
+  errorType: ApiErrorCodes.UNHANDLED_CREATE_ERROR,
+  description: ApiErrorCodeDetails.UNHANDLED_CREATE_ERROR,
+}
+
+async function _handleQuoteResponse<T = any, P extends QuoteQuery = QuoteQuery>(
+  response: Response,
+  params?: P
+): Promise<T> {
+  if (!response.ok) {
+    const errorObj: ApiErrorObject = await response.json()
+
+    // we need to map the backend error codes to match our own for quotes
+    const mappedError = mapOperatorErrorToQuoteError(errorObj)
+    const quoteError = new QuoteError(mappedError)
+
+    if (params) {
+      const { sellToken, buyToken } = params
+      log.error(`Error querying fee from API - sellToken: ${sellToken}, buyToken: ${buyToken}`)
+    }
+
+    throw quoteError
+  } else {
+    return response.json()
+  }
+}
+
+export async function getGasPrices(chainId: ChainId = DEFAULT_NETWORK_FOR_LISTS): Promise<GasFeeEndpointResponse> {
+  const response = await fetch(GAS_FEE_ENDPOINTS[chainId])
+  return response.json()
+}
```

Comment:
This method would belong maybe in a `api/safe-relay` directory. However, i think we don't need it. Its legacy endpoint